lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -0,0 +1,56 @@
+require "thread"
+
+module CC
+  module Engine
+    module Analyzers
+      class FileThreadPool
+        DEFAULT_CONCURRENCY = 2
+        MAX_CONCURRENCY = 2
+
+        def initialize(files, concurrency: DEFAULT_CONCURRENCY)
+          @files = files
+          @concurrency = concurrency
+        end
+
+        def run(&block)
+          queue = build_queue
+
+          @workers = thread_count.times.map do
+            Thread.new do
+              begin 
+                yield queue.pop(true)
+              rescue ThreadError
+              end
+            end
+          end
+        end
+
+        def join
+          workers.map(&:join)
+        end
+
+        private
+
+        attr_reader :files, :concurrency, :workers
+
+        def build_queue
+          Queue.new.tap do |queue|
+            files.each do |file|
+              queue.push(file)
+            end
+          end
+        end
+
+        def thread_count
+          if concurrency > MAX_CONCURRENCY
```

Comment:
how would you feel about using a range comparison here? http://ruby-doc.org/core-2.2.0/Range.html#method-i-3D-3D-3D

```rb
if (1..MAX_CONCURRENCY) === concurrency
  concurrency
else
  DEFAULT_CONCURRENCY
end
```